brightener.h

```diff
@@ -1,17 +1,33 @@
-#pragma once
-#include <cstdint>
-
-struct Image {
-	int rows;
-	int columns;
-	uint8_t pixels[1024 * 1024]; // max 1k x 1k image
-};
-
-class ImageBrightener {
-private:
-	Image m_inputImage;
-public:
-	ImageBrightener(Image& inputImage);
-	int BrightenWholeImage();
-	Image GetImage();
-};
+#ifndef BRIGHTENER_H
+#define BRIGHTENER_H
+
+#include <vector>
+#include <memory>
+
+class Image {
+public:
+    int rows;
+    int columns;
+    std::vector<int> pixels;
+
+    Image(int r, int c, int initialBrightness = 0)
+        : rows(r), columns(c), pixels(r* c, initialBrightness) {}
+
+    int& pixel(int x, int y) {
+        return pixels[x * columns + y];

```

Comment:
Good approach to keep the indexing logic in the `Image`. That way, the knowledge of pixel-arrangement remains encapsulated in this class itself